File: backend/ShoeStore.Domain/Entities/Orders/Order.cs
using ShoeStore.Domain.Entities.Users;

namespace ShoeStore.Domain.Entities.Orders;

public partial class Order
{
    public Guid OrderId { get; set; }

    public Guid CustomerId { get; set; }

    public Guid? EmployeeId { get; set; }

    public string Status { get; set; } = null!;

    public Guid DeliveryMethodId { get; set; }

    public decimal Subtotal { get; set; }

    public decimal Shipping { get; set; }

    public DateTime CreatedAt { get; set; }

    public virtual User Customer { get; set; } = null!;

    public virtual DeliveryMethod DeliveryMethod { get; set; } = null!;

    public virtual User? Employee { get; set; }

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/app/shared/models/orders/order.ts
import { OrderStatus } from '../../enums/order-status';
import { DeliveryMethod } from './delivery-method/delivery-method';
import { OrderItem } from './order-item';
import { User } from '../accounts/user';

export interface Order {
    orderId: string;
    customer: User;
    employee: User;
    status: string;
    deliveryMethod: DeliveryMethod;
    subtotal: number;
    shipping: number;
    createdAt: Date;
    orderItems: OrderItem[];
}